src/ReachSets/reach.jl

```diff
@@ -318,10 +317,8 @@ function d_reach(HS::HybridSystem, # Does AbstractSystem a superset of HybridSys
                         lazy_inputs_interval=kwargs_dict[:lazy_inputs_interval],
                         output_function=kwargs_dict[:output_function]
                         )
-                       for (j, tj) in enumerate(transitions(HS)) #TODO optimize it or add method for outgoing transitions to SX
-
-                            if source(HS, tj) == cur_loc_id
-                                destination_loc = HS.modes[target(HS, tj)]
+                       for j in enumerate(out_transitions(HS, cur_loc_id) #TODO optimize it or add method for outgoing transitions to SX
```

Comment:
I think you wanted to remove the `enumerate(` here (because you also changed `(j, tj)` to just `j`). Also, the comment is now outdated I think.